src/components/com_tjcertificate/administrator/controllers/certificate.php

```diff
@@ -36,6 +36,15 @@ public function download()
 		$app   = Factory::getApplication();
 		$input = $app->input;
 		$uniqueCertificateId = $input->get('certificate', '');
+		$certificate    = TJCERT::Certificate();
+
+		// Check user having permission to download
+		if (!$certificate->canDownload($uniqueCertificateId))
+		{
+			$app->enqueueMessage(Text::_('JERROR_ALERTNOAUTHOR'));
+
+			return false;
+		}
```

Comment:
Move this condition after empty condition check. This one `if (empty($uniqueCertificateId))`. Also instead of `return false` use `$app->redirect('index.php');`

Reply:
done